parity/informant.rs

```diff
@@ -190,25 +331,24 @@ impl Informant {
 				_ => String::new(),
 			},
 		);
-
-		*write_report = Some(report);
 	}
 }
 
-impl ChainNotify for Informant {
+impl ChainNotify for Informant<FullNodeInformantData> {
 	fn new_blocks(&self, imported: Vec<H256>, _invalid: Vec<H256>, _enacted: Vec<H256>, _retracted: Vec<H256>, _sealed: Vec<H256>, _proposed: Vec<Bytes>, duration: u64) {
 		let mut last_import = self.last_import.lock();
-		let sync_state = self.sync.as_ref().map(|s| s.status().state);
-		let importing = is_major_importing(sync_state, self.client.queue_info());
+		let client = &self.target.client;
+
+		let importing = self.target.is_major_importing();
 		let ripe = Instant::now() > *last_import + Duration::from_secs(1) && !importing;
 		let txs_imported = imported.iter()
 			.take(imported.len().saturating_sub(if ripe { 1 } else { 0 }))
-			.filter_map(|h| self.client.block(BlockId::Hash(*h)))
+			.filter_map(|h| client.block(BlockId::Hash(*h)))
 			.map(|b| b.transactions_count())
 			.sum();
 
 		if ripe {
-			if let Some(block) = imported.last().and_then(|h| self.client.block(BlockId::Hash(*h))) {
+			if let Some(block) = imported.last().and_then(|h| client.block(BlockId::Hash(*h))) {
 				let header_view = block.header_view();
 				let size = block.rlp().as_raw().len();
 				let (skipped, skipped_txs) = (self.skipped.load(AtomicOrdering::Relaxed) + imported.len() - 1, self.skipped_txs.load(AtomicOrdering::Relaxed) + txs_imported);
```

Comment:
how about checking this `self.skipped.load(AtomicOrdering::Relaxed) + imported.len() - 1` won't overflow or commenting why won't ever happen?